### ISSUE TITLE
Removing trees from a model

### ISSUE THREAD

**Issue by ldesreumaux** [dmlc/xgboost#5531]
As a follow up to https://github.com/dmlc/xgboost/issues/2281, I think that it would be great if XGBoost supported the removal of trees from a model. 

It would allow the implementation (with callbacks) of adaptive learning rate strategies that need to abandon some trees before adjusting the learning rate. More generally, it would allow to implement various "ensemble pruning" methods.

**Comment by trivialfis** [comment]
It's doable with `process_type` as `update`, but it's quite hacky right now.

**Comment by trivialfis** [comment]
I have a local branch that can slice the model, but it still need to handle Python's negative slice.
